lib/components/ms-slider/MsSlider.vue

```diff
@@ -0,0 +1,208 @@
+  (e: 'progressing'): void;
+  (e: 'finished'): void;
+}>();
+
+onMounted(() => {
+  addEventListener('mousemove', (e) => {
+    if (isDragging.value) {
+      updateCurrentProgress(e);
+    }
+  });
+  addEventListener('mouseup', () => {
+    if (isDragging.value) {
+      if (!atEnd()) {
+        updateSliderState(props.modelValue.progress, !needsUnpause.value);
+      }
+      isDragging.value = false;
+      needsUnpause.value = false;
+    }
+  });
+
+  if (atStart()) {
+    emits('ready');
+  }
+});
+
+onUnmounted(() => {
+  removeEventListener('mousemove', () => {});
+  removeEventListener('mouseup', () => {});
+  cancelModelValueWatch();
+  cancelPauseWatch();
+});
+
+defineExpose({
+  atStart,
+  inProgress,
+  atEnd,
+});
+
+function atStart(): boolean {
+  return props.modelValue.progress <= 0;
+}
+
+function inProgress(): boolean {
+  return props.modelValue.progress > 0 && props.modelValue.progress < props.maxValue;
+}
+
+function atEnd(): boolean {
+  return props.modelValue.progress >= props.maxValue;
+}
+
+function updateSliderState(progress: number, paused?: boolean): void {
+  emits('update:modelValue', { progress: progress, paused: paused !== undefined ? paused : props.modelValue.paused });
+}
+
+function onMouseDown(event: MouseEvent): void {
+  event.preventDefault(); // Prevent text selection on click held
+  if (props.modelValue.paused === false) {
+    updateCurrentProgress(event, true);
+    needsUnpause.value = true;
+  } else {
+    updateCurrentProgress(event);
+  }
+  isDragging.value = true;
+}
+
+function updateCurrentProgress(event: MouseEvent, paused?: boolean): void {
+  const left = sliderRef.value.getBoundingClientRect().left;
+  const right = sliderRef.value.getBoundingClientRect().right;
+  if (event.clientX < left) {
+    updateSliderState(0, paused);
+  } else if (event.clientX > right) {
+    updateSliderState(props.maxValue, paused);
+  } else {
+    updateSliderState(Math.trunc((props.maxValue * (event.clientX - left)) / (right - left)), paused);
+  }
+}
+</script>
+
+<style scoped lang="scss">
+.slider-container {
+  display: flex;
+  align-items: center;
+  gap: 0.75rem;
+
+  .slider {
+    position: relative;
+    display: flex;
+    flex-direction: column;
+    align-items: flex-start;
+    width: 100%;
+    height: 0.125rem;
+    background: var(--parsec-color-light-secondary-premiere);
+    border-radius: var(--parsec-radius-8);
+
+    .full {
+      width: v-bind(progressWidthStyle);
+      height: 0.125rem;
+      background: var(--parsec-color-light-gradient);
+      border-radius: var(--parsec-radius-6);
+    }
+
+    &-hover {
+      cursor: pointer;
+      height: 0.25rem;
+
+      .full {
+        height: 0.25rem;
+      }
+    }
+
+    .dot {
+      cursor: pointer;
+      position: absolute;
+      left: v-bind(progressWidthStyle);
+      margin-left: -0.4rem;
+      height: 0.8rem;
+      width: 0.8rem;
+      -ms-transform: translateY(-35%);
+      transform: translateY(-35%);
```

Comment:
Not sure it is necessary, `transform` property is available on every browsers (https://caniuse.com/?search=transform)